src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/components/options_list_editor_options.test.tsx

```diff
@@ -92,24 +99,6 @@ describe('Options list sorting button', () => {
   });
 
   describe('custom search options', () => {
-    test('do not show custom search options when `allowExpensiveQueries` is false', async () => {
-      const allowExpensiveQueries$ = new BehaviorSubject<boolean>(false);
-      const controlGroupApi = getMockedControlGroupApi(undefined, { allowExpensiveQueries$ });
-      const component = mountComponent({
-        initialState: getMockedState(),
-        field: { type: 'string' } as DataViewField,
-        controlGroupApi,
-      });
-      expect(
-        component.queryByTestId('optionsListControl__searchOptionsRadioGroup')
-      ).not.toBeInTheDocument();
-
-      act(() => allowExpensiveQueries$.next(true));
-      expect(
-        component.queryByTestId('optionsListControl__searchOptionsRadioGroup')
-      ).toBeInTheDocument();
-    });
```

Comment:
Discussed with @ThomThomson offline, and we decided it was okay for the editor to display "incompatible" search techniques  due to the effort of implementing this. We also will be [removing the `allowExpensiveQueries` very shortly](https://github.com/elastic/kibana/issues/243705) so it wasn't worth implementing this in the short term